picasso/src/main/java/com/squareup/picasso/OkHttp3Downloader.java

```diff
@@ -32,6 +33,7 @@
 public final class OkHttp3Downloader implements Downloader {
   private final Call.Factory client;
   private final Cache cache;
+  private boolean sharedClient = true;
```

Comment:
I'm not sure how to do that actually in a fall through constructor like this. The current structure is that the second-to-last constructor knows it's making a default okhttpclient instance. The final constructor would need some way of knowing if the client it's given was created internally or not